### DISCUSSION TITLE
My experience with Copilot -> bad! really bad!

### DISCUSSION THREAD

**Discussion by Kingside88** [community/community#45659] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Yes we are all fixed by ChatGPT and are very excited about AI.
Having ChatGPT inside your're IDE like VS Code or VS sounds great, right?
But Copilot is far away from beeing great.
First try was with VS Code. After installation I could not connect with GitHub with the error everybody knows, when he is using this extension: `GitHub Copilot could not connect to server. Extension activation failed: "Timed out waiting for authentication provider to register"`
Yeah great. After a long research I could fix it by uninstall all and delete the whole AppData from VS Code. Great. After a clean install the extension could connect to the server.
In my nuxt 3 application it starts showing me some suggestions. But the suggestions were wrong. Wrong configuration entries and wrong code as itself.
This was some days ago. Today I open up VS Code and received the same error: `GitHub Copilot could not connect to server...`

My test with Visual Studio is different.
It could connect from minute one. And the first try for creating unit tests was impressive.
But followed by stops working. After few minutes it just stops working without any output error.
Restarting Visual studio fix this problem again for few minutes.
And it generates code which is not finished. Just a cut in the middle of a method. Very annoying.
When you enter some comments for creating something for you, it sometimes work and sometimes not. Correcting the comment results to the same suggestion as before correction.

In this circumstances I will not subscribe and pay Microsoft 100$ anually . Never ever. Please share youre experience

**Comment by BenGardiner123** [comment]
Yes that is my experience as well. It's quite broken

**Comment by mfirthOmx** [comment]
When I started out it was great: then it went to absolutely terrible. It wastes so much time with bad incomplete and non working suggestions.

**Reply by ayim** [reply]
@mfirthOmx sorry to hear the tool hasn't been helpful for you.
Would you be willing to hop on a call with us to better understand exactly what's going on with your instance of Copilot?

--
Aaron
Product Manager for GitHub Copilot in Visual Studio
aayim@microsoft.com

**Comment by destinformore** [comment]
pure trash

**Reply by ayim** [reply]
@destinformore if you're having issues with the Visual Studio instance, we'd love to hop on a call and debug with ya.

--
Aaron
Product Manager for GitHub Copilot in Visual Studio
[aayim@microsoft.com](mailto:aayim@microsoft.com)

**Comment by Jamshibkl** [comment]
same here!

**Reply by ayim** [reply]
@Jamshibkl if you're having issues with the Visual Studio instance, let's hop on a call and get this working for ya.

--
Aaron
Product Manager for GitHub Copilot in Visual Studio
[aayim@microsoft.com](mailto:aayim@microsoft.com)

**Comment by RaduTudorIon** [comment]
Talking about copilot chat, sometimes copilot "crashes" on visual studio, it simply doesn't deliver a response, also you can't copy paste from the response window and it's a frustrating user experience.

**Reply by ayim** [reply]
@rheapatel23

**Reply by rheapatel23** [reply]
Hi @RaduTudorIon, sorry to hear that isn't working for you. Would you be willing to share more about this over a call? If so please leave your email 

Thanks, 
Rhea -Product Manager for GitHub Copilot Visual Studio

**Comment by mahdimohamadpoor** [comment]
its a new feature can be better in future

**Comment by RaduTudorIon** [comment]
Now not only I had a pretty bad experience using copilot in visual studio recently, but today I also got logged out with a message " Your GitHub Copilot subscription has ended. "! After not changing anything to billing information or any settings. Re-entering the previous billing information solved the issue, it seems that it stopped billing my card since december.

**Comment by Tape-Worm** [comment]
Up until today, it's been brilliant for generating tests (I despise writing tests, so this is a huge thing for me). But today, using the same prompts, it's ... I don't know... stupid?  

For example: I told it to write tests for my WriteByte method. Sure, no problem, it generates the tests as I expect, and as usual, it wraps the methods in a class declaration (this is normal). So, I ask it to "remove the class declaration from the tests". Yesterday, when I asked it to do that, it did so without fail. Today, it started telling me stuff about what I could do with my WriteByte method.

Another example: (you folks will f***ing love this one) I start a new conversation and say, "For this conversation, we're going to use C# 12, and .NET 8" so it writes things using the latest syntax. Again, yesterday, no problem, did it like a champ. Today this is what it told me:
![image](https://github.com/community/community/assets/9710137/7c6f325c-fbc2-4f8a-8b61-6cd44450e3de)

What in all of creation is going on here!??  .NET 5 was put out to pasture ages ago! Sure, I can see that if it was trained on .NET 5 and lower... but that seems kind of dumb for a coding assistant to use something as far back as C# 9... This is so confusing and very disappointing.

At this rate, this is not **_NOT_** worth $100/year.

**Reply by Tape-Worm** [reply]
Oh I should point out that I'm using Copilot Chat on VS 2022.

**Reply by ayim** [reply]
@Tape-Worm yoh! thanks for reaching out. I mostly support the autocomplete product but @rheapatel23's gonna look at this on Monday.

--
Aaron Yim
Product Manager for GitHub Copilot in Visual Studio
github: ayim | aayim@microsoft.com

**Reply by Tape-Worm** [reply]
Thanks @ayim 

I'm using it today and it seems to be working better, so whatever was going on yesterday seems like it might have been a hiccup.  

I wish I could get this thing to save my "preferences", that is when I tell it to use C# 12, .NET 8, MS test framework (always defaults to nunit for some reason), etc... I wish I could get it to write a local file that I can reload on chat initialization (or better yet, on each query).  Would make life a bit easier.

**Reply by ayim** [reply]
CC @rheapatel23 ^

**Comment by RaduTudorIon** [comment]
![image](https://github.com/community/community/assets/15893272/b6e162f9-a7a8-4a56-9afc-d85cc3348a1b)
Copilot for Visual Studio 2022 when asking a question about a popular .net class https://learn.microsoft.com/en-us/dotnet/api/system.linq.enumerable?view=net-8.0

**Reply by ayim** [reply]
@rheapatel23 for tracking RAI related redactions

**Reply by ayim** [reply]
@RaduTudorIon 
I wonder if there was something in your code that hit a word filter.
I tried to repro a couple ways and couldn't repro.

**Reply by RaduTudorIon** [reply]
@ayim I think I wrote the prompt in this repo, started it with a .net web api template project, nothing weird in the project setup https://github.com/RaduTudorIon/FomulaOneApi

**Reply by RaduTudorIon** [reply]
@ayim Selected the method and used this prompt
![image](https://github.com/community/community/assets/15893272/f8f50c35-6882-43a7-8953-a3d07650003e)

**Reply by ayim** [reply]
Could you file a ticket using the "Report a PRoblem" mechanism?
It'll include relevant logs for us to diagnose.

![image](https://github.com/community/community/assets/10951869/14ba5879-1be4-49e5-b441-ce07e3e62f8b)

**Reply by RaduTudorIon** [reply]
@ayim submitted, thank you

**Reply by ayim** [reply]
Thanks, going to close and consolidate our chat on the VS Feedback side of things.

**Comment by bastienJS** [comment]
Some minimum requirements I have but copilot is not able to do it:

1. Copilot please compare all properties of the expectedProfile with the actualProfile => What I got are some sample comparisons with fantasy properties HAHAHA
2. Copilot please initialize all properties of _profile instance => What I got were some initialized Lists but not the simple properties like string, int etc... OMG

**Reply by ayim** [reply]
@bastienJS have you tried using "#" to add relevant symbols/classes/files to your query?
It should improve the context.
We're also thinking about symbolic information, so keep an eye out.
